In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
data = {
    "Hours Studies": [2, 3, 4, 5, 6, 7],
    "Hours Watching TV": [4, 3, 2, 1, 0, 0],
    "Outdoor Activity Time": [2, 4, 6, 8, 10, 12],
    "Hours Listening to Music": [2, 3, 4, 1, 5, 0],
    "Water Consumed": [5, 6, 5, 6, 4, 5],
    "Test Score": [65, 70, 75, 80, 85, 90]
}
df = pd.DataFrame(data)
df

Hours Studies  Hours Watching TV  Outdoor Activity Time  \
0              2                  4                      2   
1              3                  3                      4   
2              4                  2                      6   
3              5                  1                      8   
4              6                  0                     10   
5              7                  0                     12   

   Hours Listening to Music  Water Consumed  Test Score  
0                         2               5          65  
1                         3               6          70  
2                         4               5          75  
3                         1               6          80  
4                         5               4          85  
5                         0               5          90

In [3]:
def correlation_coeff(x, y):
    x_mean, y_mean = np.mean(x), np.mean(y)
    x_std, y_std = np.std(x), np.std(y)
    n = len(x)
    return np.sum((x-x_mean)*(y-y_mean)) / (n*x_std*y_std)

In [16]:
n =len(df.columns)
corr_test = []
for i in range(n-1):
    corr_test.append(correlation_coeff(df[df.columns[i]],df[df.columns[n-1]]))

corr_test = pd.DataFrame(corr_test,columns=['Test Score'],index=df.columns[:-1])
corr_test


Test Score
Hours Studies               1.000000
Hours Watching TV          -0.981981
Outdoor Activity Time       1.000000
Hours Listening to Music   -0.200000
Water Consumed             -0.355036

In [24]:
pos,neg,non = [],[],[]
for i in range(len(corr_test)):
    value = round(corr_test.iloc[i,0],2)
    if value > 0.5:
        pos.append(f"{corr_test.index[i]} - {corr_test.columns[0]} : {value}")                                
    elif value < -0.5:
        neg.append(f"{corr_test.index[i]} - {corr_test.columns[0]} : {value}")
    else:
        non.append(f"{corr_test.index[i]} - {corr_test.columns[0]} : {value}")
print("Positive Correlation")
for i in pos:
    print(i)
print("\nNegative Correlation")
for i in neg:
    print(i)
print("\nNon Correlation")
for i in non:
    print(i)

Positive Correlation
Hours Studies - Test Score : 1.0
Outdoor Activity Time - Test Score : 1.0

Negative Correlation
Hours Watching TV - Test Score : -0.98

Non Correlation
Hours Listening to Music - Test Score : -0.2
Water Consumed - Test Score : -0.36


## Positive Correlations

Hours Studied: The direct correlation between studying and improving one's understanding of the material can be utilised to explain the positive correlation between the quantity of study time and exam performance. More time spent studying aids in-depth learning, problem-solving practise, and content familiarisation, all of which enhance exam performance.

Outdoor Activity(+1): There is a positive correlation between participation in outdoor activities and test scores, indicating that doing so may improve pupils' academic performance. Exercise has been shown to boost cognitive function, emotional control, and overall wellbeing, all of which can enhance concentration and academic achievement.


## Negative Correlation

TV viewing time (-0.98): When compared to studying and other beneficial activities, there is a substantial negative relationship between the amount of time spent watching TV and test scores, which underscores the potential time and attention losses associated with excessive television use. The importance of prioritising learning and studying while efficiently managing your time and avoiding distractions is highlighted by this.



## No Significance Correlations

Hours listening to music (-0.2): Although not statistically significant, the modestly favourable relationship between the amount of time listening to music and test scores implies that music may have a minor positive impact on concentration and relaxation during study sessions. Finding a balance that works for one's learning preferences and concentration needs is essential, but it's necessary to keep in mind that everyone has different preferences for studying environments.

Water Consumed (0.36): Although not statistically significant, the marginally positive connection between the amount of water consumed and test outcomes suggests the potential that staying well hydrated may improve cognitive performance. Staying hydrated is essential to avoid focus issues, fatigue, and a loss of mental alertness, all of which can have a detrimental impact on exam performance.

These connections shed light on the complex relationships between several variables and academic achievement, emphasising the importance of efficient study techniques, conscientious time management, and general wellbeing.